lib/rules/no-array-destructured-return.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview Disallow array destructuring of return values.
+ * @author Vitor Balocco
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow array destructuring of return values",
+            category: "Best Practices",
+            recommended: false
+        },
+
+        schema: []
+    },
+
+    create(context) {
+
+        /**
+         * Reports an error on the given node.
+         * @param   {ASTNode} node The node to report on.
+         * @returns {void}
+         */
+        function report(node) {
+            context.report({
+                node,
+                message: "Array destructuring of return values is not allowed, use object destructuring instead."
```

Comment:
The ship has probably already sailed for this, but why is the rule warning for

```js
const [foo, bar] = baz();
```

rather than

```js
return [foo, bar]
```

? It seems like it's trying to prevent the pattern in the second example, but it's doing so indirectly by banning the first pattern.